packages/veraswap-app/src/atoms/token-balance.ts

```diff
@@ -496,3 +500,25 @@ export const tokenOutUsdValueAtom = atom((get) => {
     const amountOutBigInt = BigInt(parseFloat(amountOut) * 10 ** currencyOut.decimals);
     return calculateUsdValueFromQuote(amountOutBigInt, quote, usdDecimals, currencyOut.chainId);
 });
+
+export const tokenInAllowanceAccountToPoolQueryAtom = atom((get) => {
+    const currency = get(currencyInAtom);
+    const account = get(accountAtom);
+    const stargateQuote = get(stargateQuoteAtom).data;
+
+    if (!currency || !account?.address || !stargateQuote || stargateQuote.type !== "TOKEN") {
```

Comment:
I think you should rely on the `transactionType`. Like in the stargate quote.
This is only relevant if doing a simple bridge
(iirc, for bridge and swap, we check permissions in getTransaction)
Should we maybe check permit2 permission here? or just stick to regular permit?